packages/peregrine/lib/talons/Image/useResourceImage.js

```diff
@@ -40,38 +40,44 @@ export const useResourceImage = props => {
     // Example: 100px
     // Example: (max-width: 640px) 50px, 100px
     const sizes = useMemo(() => {
-        // The values in resourceSizes are numbers. Convert to string by adding 'px'.
-        const getPixelSize = sizeName => resourceSizes.get(sizeName) + 'px';
+        // The values in widths are numbers. Convert to string by adding 'px'.
+        const getPixelSize = index => widths[index] + 'px';
 
         // Helper function for prepending sizes media constraints.
-        const constrain = sizeName => {
-            const breakpoint = resourceSizeBreakpoints.get(sizeName) + 'px';
-            const size = getPixelSize(sizeName);
+        const constrain = index => {
+            const breakpoint = widthBreakpoints[index] + 'px';
+            const size = getPixelSize(index);
 
             return `(max-width: ${breakpoint}) ${size}`;
         };
 
-        // Note: it is assumed sizes will be filled from small, to medium, to large.
-        // In other words, having values for small and large but not medium is not supported.
-        switch (resourceSizeBreakpoints.size) {
-            case 2: {
-                const small = constrain('small');
-                const medium = constrain('medium');
-                const large = getPixelSize('large');
+        // The number of breakpoints must be one less than the number of sizes.
```

Comment:
There are a few "hidden" assumptions within this PR:

1. The number of breakpoints will be one less than the number of widths

There needs to be one width that is unconstrained. In other words, one that doesn't have a media query.

**Bad**:
```js
const breakpoints = [100, 200];
const widths = [50, 150];

// sizes = "(max-width: 100px) 50px, (max-width: 200px) 150px"
```

This is not a valid value for `sizes`.

**Good**:
```js
const breakpoints = [100];
const widths = [50, 150];

// sizes = "(max-width: 100px) 50px, 150px"
```

❓ Should I add logic to enforce this constraint? Basically, only use `widths.length - 1` `widthBreakpoints`?

2. The values in `widths` and `widthBreakpoints` will be sorted from small to large

This is because we hardcode `max-width` for the `sizes` media queries.

**Bad**:
```js
const breakpoints = [200, 100];
const widths = [250, 350, 150];

// sizes = "(max-width: 200px) 250px, (max-width: 100px) 350px, 150px".
```

This is a valid value for `sizes` but isn't going to be what the developer expects. In English this reads: "up to 100px, use the 250px-wide image. Between 100 and 200px, use the 350px-wide image. Beyond 200px, use the 150px image". 

**Good**:
```js
const breakpoints = [100, 200];
const widths = [150, 250, 350];

// sizes = "(max-width: 100px) 150px, (max-width: 200px) 250px, 350px".
```

❓  Should we sort them if they're not already sorted?

3. The entries in widths and breakpoints "match up"

That is, `breakpoint[0]` is for `widths[0]`.

```js
const breakpoints = [100];
const widths = [150, 250];

// sizes = "(max-width: 100px) 150px, 250px"
```

Reply:
> Should I add logic to enforce this constraint?

What about an `unconstrainedWidth` or some other name representing it? 

> The values in widths and widthBreakpoints will be sorted from small to large

They *should* be sorted right? I'm starting to think an array might be the wrong structure. Either that or we would have to add some prop validation for it.

Reply:
> This is a valid value for `sizes` but isn't going to be what the developer expects. In English this reads: "up to 100px, use the 250px-wide image. Between 100 and 200px, use the 350px-wide image. Beyond 200px, use the 150px image".
> 
> ❓ Should we sort them if they're not already sorted?

I don't think we should sort them. We should leave this to the developer because 

```es6
const breakpoints = [200, 100];
const widths = [250, 350, 150];

// sizes = "(max-width: 200px) 250px, (max-width: 100px) 350px, 150px".
```

is a valid use case.

Potential Usecase:

In a desktop view, the product image should be shown on the left with details on right. Which means 50% of the width.

In a mobile view, the product image should take up the whole width. Which means 100% of the width.

This means, as the width of the viewport goes down, the size of the image should go up, hence it should download a high res image.

If we sort the `widths` array, this won't be possible.

This is replicable in Venia's product page as well:

Desktop View:
![image](https://user-images.githubusercontent.com/35203638/68139098-2b621d80-feef-11e9-9bf7-0f65818fbe49.png)


iPhone View:
![image](https://user-images.githubusercontent.com/35203638/68139052-171e2080-feef-11e9-902f-6cdc849c60b7.png)

Reply:
Resolved by https://github.com/magento/pwa-studio/pull/1956/commits/fa254e23d670ce5fa1f3e2d1aac25ccdfaff2562.